packages/sit-onyx/src/components/OnyxButton/OnyxButton.vue

```diff
@@ -23,11 +25,12 @@ const emit = defineEmits<{
     v-else
     class="onyx-button"
     :class="[`onyx-button--${props.variation}`, `onyx-button--${props.mode}`]"
-    :disabled="props.disabled"
+    :disabled="props.disabled || props.loading"
     @click="emit('click')"
   >
-    <OnyxIcon v-if="props.icon" :icon="props.icon" size="24px" />
-    <span class="onyx-button__label onyx-truncation-ellipsis">{{ props.label }}</span>
+    <OnyxIcon v-if="props.icon && !props.loading" :icon="props.icon" size="24px" />
+    <OnyxLoadingIndicator v-if="props.loading" class="onyx-button__loading" type="dots" />
+    <span v-else class="onyx-button__label onyx-truncation-ellipsis">{{ props.label }}</span>
```

Comment:
```suggestion
    <OnyxIcon v-if="props.icon && !props.loading" :icon="props.icon" />
    <OnyxLoadingIndicator v-if="props.loading" class="onyx-button__loading" />
    <span v-else class="onyx-button__label onyx-truncation-ellipsis">{{ props.label }}</span>
```

We don't need to set the props if they are the default